### ISSUE TITLE
ares v130 crashes on macOS when loading Nintendo 64 ROMs

### ISSUE THREAD

**Issue by command-tab** [higan-emu/ares#816]
**Describe the bug**
When loading a Nintendo 64 ROM, ares crashes instantly. This does not happen under ares v129.

**To Reproduce**
1. Select Load → Nintendo → Nintendo 64... from the menu bar
2. Choose a standard USA z64 ROM (e.g. _Super Mario 64_ or _Goldeneye 007_)
3. Ares crashes and macOS shows the standard application crash alert dialog

**Expected behavior**
I expected ares to load the ROM (even if there are gameplay or rendering issues).

**Additional context**
macOS 12.6, x86_64

<details>
  <summary>Crash Log</summary>

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               ares [30429]
Path:                  /private/var/folders/*/ares.app/Contents/MacOS/ares
Identifier:            dev.ares.ares
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               5406

Date/Time:             2022-09-30 13:26:00.4184 -0700
OS Version:            macOS 12.6 (21G115)
Report Version:        12
Bridge OS Version:     6.6 (19P6067)
Anonymous UUID:        908AC9B2-F72B-FBE8-E346-DBEEC2209C70

Sleep/Wake UUID:       5DF6C8CD-ACE0-4F8C-B2A0-506541D35C57

Time Awake Since Boot: 490000 seconds
Time Since Wake:       19615 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x000000010632b210
Exception Codes:       0x0000000000000002, 0x000000010632b210
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 10 Bus error: 10
Terminating Process:   exc handler [30429]

VM Region Info: 0x10632b210 is in 0x106317000-0x10e317000;  bytes after start: 82448  bytes before end: 134135279
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      __DATA                      106107000-106317000    [ 2112K] rw-/rwx SM=COW  ...ts/MacOS/ares
--->  __DATA                      106317000-10e317000    [128.0M] rw-/rwx SM=PRV  ...ts/MacOS/ares
      __DATA                      10e317000-116317000    [128.0M] rw-/rwx SM=PRV  ...ts/MacOS/ares

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   ???                           	       0x10632b210 ares::Memory::fixedBuffer + 528

Thread 1:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 2:: AMCP Logging Spool
0   libsystem_kernel.dylib        	    0x7ff809ade9b6 semaphore_wait_trap + 10
1   caulk                         	    0x7ff81261b2e6 caulk::mach::semaphore::wait_or_error() + 16
2   caulk                         	    0x7ff812603148 caulk::concurrent::details::worker_thread::run() + 36
3   caulk                         	    0x7ff812602e0c void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 41
4   libsystem_pthread.dylib       	    0x7ff809b1b4e1 _pthread_start + 125
5   libsystem_pthread.dylib       	    0x7ff809b16f6b thread_start + 15

Thread 3:: com.apple.audio.IOThread.client
0   libsystem_kernel.dylib        	    0x7ff809ade97a mach_msg_trap + 10
1   libsystem_kernel.dylib        	    0x7ff809adece8 mach_msg + 56
2   CoreAudio                     	    0x7ff80b761c07 HALB_MachPort::SendSimpleMessageWithSimpleReply(unsigned int, unsigned int, int, int&, bool, unsigned int) + 111
3   CoreAudio                     	    0x7ff80b5ec729 HALC_ProxyIOContext::IOWorkLoop() + 3931
4   CoreAudio                     	    0x7ff80b5eb205 invocation function for block in HALC_ProxyIOContext::HALC_ProxyIOContext(unsigned int, unsigned int) + 63
5   CoreAudio                     	    0x7ff80b7b7616 HALB_IOThread::Entry(void*) + 72
6   libsystem_pthread.dylib       	    0x7ff809b1b4e1 _pthread_start + 125
7   libsystem_pthread.dylib       	    0x7ff809b16f6b thread_start + 15

Thread 4:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	    0x7ff809ade97a mach_msg_trap + 10
1   libsystem_kernel.dylib        	    0x7ff809adece8 mach_msg + 56
2   SkyLight                      	    0x7ff80eaeff6c CGSSnarfAndDispatchDatagrams + 152
3   SkyLight                      	    0x7ff80edb6e5e SLSGetNextEventRecordInternal + 258
4   SkyLight                      	    0x7ff80ec026fd SLEventCreateNextEvent + 9
5   HIToolbox                     	    0x7ff81289c039 PullEventsFromWindowServerOnConnection(unsigned int, unsigned char, __CFMachPortBoost*) + 45
6   HIToolbox                     	    0x7ff81289bfd5 MessageHandler(__CFMachPort*, void*, long, void*) + 48
7   CoreFoundation                	    0x7ff809c0f580 __CFMachPortPerform + 250
8   CoreFoundation                	    0x7ff809be2924 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 41
9   CoreFoundation                	    0x7ff809be2804 __CFRunLoopDoSource1 + 619
10  CoreFoundation                	    0x7ff809be0e6b __CFRunLoopRun + 2415
11  CoreFoundation                	    0x7ff809bdfe3c CFRunLoopRunSpecific + 562
12  AppKit                        	    0x7ff80c7879ce _NSEventThread + 132
13  libsystem_pthread.dylib       	    0x7ff809b1b4e1 _pthread_start + 125
14  libsystem_pthread.dylib       	    0x7ff809b16f6b thread_start + 15

Thread 5:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 6:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 7:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 8::  Dispatch queue: com.apple.SkyLight.UpdateManager.SerialSema
0   libsystem_kernel.dylib        	    0x7ff809ade9b6 semaphore_wait_trap + 10
1   SkyLight                      	    0x7ff80eb4533f invocation function for block in (anonymous namespace)::InterprocessSerialSemaphore::asyncWait(void (int) block_pointer) + 35
2   libdispatch.dylib             	    0x7ff8099630cc _dispatch_call_block_and_release + 12
3   libdispatch.dylib             	    0x7ff809964317 _dispatch_client_callout + 8
4   libdispatch.dylib             	    0x7ff80996a317 _dispatch_lane_serial_drain + 672
5   libdispatch.dylib             	    0x7ff80996adfd _dispatch_lane_invoke + 366
6   libdispatch.dylib             	    0x7ff809974eee _dispatch_workloop_worker_thread + 753
7   libsystem_pthread.dylib       	    0x7ff809b17fd0 _pthread_wqthread + 326
8   libsystem_pthread.dylib       	    0x7ff809b16f57 start_wqthread + 15

Thread 9:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 10::  Dispatch queue: NSCGSDisableUpdates
0   libsystem_kernel.dylib        	    0x7ff809ade9ce semaphore_timedwait_trap + 10
1   libdispatch.dylib             	    0x7ff809964848 _dispatch_sema4_timedwait + 72
2   libdispatch.dylib             	    0x7ff809964c75 _dispatch_semaphore_wait_slow + 58
3   SkyLight                      	    0x7ff80eb4512e CGSUpdateManager::enable_updates_common() + 1038
4   SkyLight                      	    0x7ff80eb45b7e SLSReenableUpdateTokenWithSeed + 130
5   libdispatch.dylib             	    0x7ff8099630cc _dispatch_call_block_and_release + 12
6   libdispatch.dylib             	    0x7ff809964317 _dispatch_client_callout + 8
7   libdispatch.dylib             	    0x7ff80996a317 _dispatch_lane_serial_drain + 672
8   libdispatch.dylib             	    0x7ff80996adfd _dispatch_lane_invoke + 366
9   libdispatch.dylib             	    0x7ff809974eee _dispatch_workloop_worker_thread + 753
10  libsystem_pthread.dylib       	    0x7ff809b17fd0 _pthread_wqthread + 326
11  libsystem_pthread.dylib       	    0x7ff809b16f57 start_wqthread + 15

Thread 11:
0   libsystem_pthread.dylib       	    0x7ff809b16f48 start_wqthread + 0

Thread 12:
0   libsystem_kernel.dylib        	    0x7ff809ae12be __semwait_signal + 10
1   libsystem_c.dylib             	    0x7ff8099f5863 nanosleep + 196
2   libsystem_c.dylib             	    0x7ff8099f5799 usleep + 53
3   ares                          	       0x1050888f5 ares::Core::Video::Screen::main(unsigned long) + 37
4   libsystem_pthread.dylib       	    0x7ff809b16f6b thread_start + 15

Thread 13:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d93 std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
3   ares                          	       0x1053bd1d1 osd_event::wait(unsigned long long) + 321

Thread 14:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d93 std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
3   ares                          	       0x1053bd1d1 osd_event::wait(unsigned long long) + 321

Thread 15:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d93 std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
3   ares                          	       0x1053bd1d1 osd_event::wait(unsigned long long) + 321

Thread 16:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d02 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
3   ares                          	       0x105305c75 RDP::WorkerThread<RDP::CoherencyOperation, RDP::CommandProcessor::FenceExecutor>::main_loop() + 181
4   ares                          	       0x105306199 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (RDP::WorkerThread<RDP::CoherencyOperation, RDP::CommandProcessor::FenceExecutor>::*)(), RDP::WorkerThread<RDP::CoherencyOperation, RDP::CommandProcessor::FenceExecutor>*> >(void*) + 57
5   libsystem_pthread.dylib       	    0x7ff809b16f6b thread_start + 15

Thread 17:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d02 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18

Thread 18:
0   libsystem_kernel.dylib        	    0x7ff809ae13ea __psynch_cvwait + 10
1   libsystem_pthread.dylib       	    0x7ff809b1ba6f _pthread_cond_wait + 1249
2   libc++.1.dylib                	    0x7ff809a79d93 std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
3   ares                          	       0x1053006a9 std::__1::cv_status std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > const&) + 169
4   ???                           	    0x70000b0e9000 ???


Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x000000010632b210  rbx: 0x000000010e606430  rcx: 0x000000010632b000  rdx: 0x000000010e610070
  rdi: 0x000000010e606430  rsi: 0x000000010e60fe68  rbp: 0x0000000000000000  rsp: 0x00007ff7baf78fc8
   r8: 0x0000000000000140   r9: 0x00000000000000a0  r10: 0x00000000000007fb  r11: 0x0000000000000065
  r12: 0x00007fc6fc907cf8  r13: 0x00007ff8099ae270  r14: 0x000000010e606430  r15: 0x00007ff8244f04d7
  rip: 0x000000010632b210  rfl: 0x0000000000010202  cr2: 0x000000010632b210
  
Logical CPU:     0
Error Code:      0x00000015 (invalid protections for user instruction read)
Trap Number:     14


Binary Images:
               0x0 - 0xffffffffffffffff ??? (*) <00000000-0000-0000-0000-000000000000> ???
    0x7ff809b15000 -     0x7ff809b20fff libsystem_pthread.dylib (*) <b5454e27-e8c7-3fdb-b77f-714f1e82e70b> /usr/lib/system/libsystem_pthread.dylib
    0x7ff809add000 -     0x7ff809b14fff libsystem_kernel.dylib (*) <8cc28466-fd2f-3c80-9834-9525b7beac19> /usr/lib/system/libsystem_kernel.dylib
    0x7ff812601000 -     0x7ff812622fff com.apple.audio.caulk (1.0) <8e7b3d95-1d47-3f17-9512-c5fcc30792c2> /System/Library/PrivateFrameworks/caulk.framework/Versions/A/caulk
    0x7ff80b432000 -     0x7ff80bb67fff com.apple.audio.CoreAudio (5.0) <9be547d0-0af1-3470-afde-2ac4cd34441e> /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
    0x7ff80eae6000 -     0x7ff80ee84fff com.apple.SkyLight (1.600.0) <6563e632-526d-3c88-8035-049fcca8ca93> /System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/SkyLight
    0x7ff812861000 -     0x7ff812b54fff com.apple.HIToolbox (2.1.1) <06fdecd6-9f69-397b-b1e2-a8226c0ba7ed> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
    0x7ff809b62000 -     0x7ff80a064fff com.apple.CoreFoundation (6.9) <93c48919-68af-367e-9a67-db4159bc962c> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7ff80c5db000 -     0x7ff80d46afff com.apple.AppKit (6.9) <06015263-62ac-3b08-a298-dc835c18452a> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
    0x7ff809961000 -     0x7ff8099a7fff libdispatch.dylib (*) <1a04b380-76e4-3e4b-b0fc-9837533d021d> /usr/lib/system/libdispatch.dylib
    0x7ff8099e5000 -     0x7ff809a6dfff libsystem_c.dylib (*) <e42e9d7a-03b4-340b-b61e-dcd45fd4acc0> /usr/lib/system/libsystem_c.dylib
       0x104f83000 -        0x106106fff dev.ares.ares (*) <d332ee49-9690-3e21-80af-1e90fc8ea4dc> /private/var/folders/*/ares.app/Contents/MacOS/ares
    0x7ff809a6e000 -     0x7ff809ac6fff libc++.1.dylib (*) <e67ac91f-f1dc-30ec-84f3-83f5bdad3434> /usr/lib/libc++.1.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 258
    thread_create: 0
    thread_set_state: 24639

VM Region Summary:
ReadOnly portion of Libraries: Total=1.0G resident=0K(0%) swapped_out_or_unallocated=1.0G(100%)
Writable regions: Total=1.4G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=1.4G(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Accelerate framework               512K        4 
Activity Tracing                   256K        1 
CG backing stores                 3840K        4 
CG image                           816K       14 
ColorSync                          220K       26 
CoreAnimation                     9292K       31 
CoreGraphics                        12K        2 
CoreUI image data                 1972K       14 
Foundation                          16K        1 
Kernel Alloc Once                    8K        1 
MALLOC                           747.2M      121 
MALLOC guard page                   32K        7 
MALLOC_NANO (reserved)           256.0M        1         reserved VM address space (unallocated)
ObjC additional data                15K        1 
OpenGL GLSL                        256K        3 
STACK GUARD                       56.1M       19 
Stack                             17.1M       19 
VM_ALLOCATE                        992K       33 
__CTF                               756        1 
__DATA                           471.6M      385 
__DATA_CONST                      19.7M      229 
__DATA_DIRTY                       875K      128 
__FONT_DATA                          4K        1 
__GLSLBUILTINS                    5176K        1 
__LINKEDIT                       653.7M       16 
__TEXT                           366.5M      393 
__UNICODE                          592K        1 
dyld private memory               1024K        1 
mapped file                      165.4M       29 
shared memory                      856K       19 
===========                     =======  ======= 
TOTAL                              2.7G     1506 
TOTAL, minus reserved VM space     2.5G     1506
```
</details>

**Comment by LukeUsher** [comment]
I can’t produce this locally, works great on my Mac running Catalina. 

Unfortunately I can’t test on newer versions as my hardware is too old to run them.

**Comment by jamiet83** [comment]
Yep I get the same error.

iMac 27 inch 5k from 2017, latest macos

**Comment by LukeUsher** [comment]
It seems the reason I can't reproduce this is because Catalina doesn't support the higher security model that later versions of macOS do.

We incorrectly assumed that some of the requirements were exclusive to Apple Silicon, but they impact the latest versions macOS on Intel too.

We're working on a solution.

**Comment by command-tab** [comment]
Fixed by 3c99afc and 66b60d6 🙌🏻

**Comment by LukeUsher** [comment]
A point-release v130.1 is being prepared as a hotfix.
